chore(devextreme-react): replace Gulp with NX plugin

chaosmirage opened the pull request:
<!--
Before you submit a pull request, review our contribution guidelines (CONTRIBUTING.md).

If your pull request contains a bug fix, its title should include "fixes ISSUE_ID" or "resolves ISSUE_ID".

In addition, please clarify:

1. What does the PR change?
2. How did you achieve this?
3. How can we verify these changes?

-->

chaosmirage committed:
chore(devextreme-react): remove gulp

chaosmirage committed:
chore(devextreme-react): replace gulp with nx plugin

chaosmirage committed:
chore: add prettier

chaosmirage commented on packages/nx-infra-plugin/prod/package.json:
```diff
@@ -0,0 +1,27 @@
+    "build": "nx build nx-infra-plugin",
+    "test": "nx test nx-infra-plugin"
+  },
+  "main": "./src/index.js"
```
expected

chaosmirage commented on packages/nx-infra-plugin/prod/src/executors/generate-react-components/executor.js:
```diff
@@ -0,0 +1,225 @@
+const GENERATE_REEXPORTS = 'generateReexports';
+const GENERATE_CUSTOM_TYPES = 'generateCustomTypes';
+const QUOTES_DOUBLE = 'double';
+const EXPLICIT_INDEX_IN_IMPORTS = 'excplicitIndexInImports';
```
checked, it is misspell in the API (it works)

chaosmirage committed:
chore: return `silent: false`

alexslavr commented on packages/nx-infra-plugin/package.json:
```diff
@@ -0,0 +1,32 @@
+  "scripts": {
+    "format:check": "prettier --check .",
+    "format": "prettier --write .",
+    "build": "nx build nx-infra-plugin",
```
Running `pnpm run build` from workspace dir `packages/nx-infra-plugin` will fail with `NX Cannot find project 'nx-infra-plugin'`. Because NX project has different name with `devextreme-*` prefix. Though it works fine from repo root `nx run devextreme-infra-plugin:build`

chaosmirage replied:
fixed

alexslavr commented on packages/nx-infra-plugin/prod/package.json:
```diff
@@ -0,0 +1,33 @@
+{
```
Have you considered running `pnpm run build` on postinstall?

chaosmirage replied:
added via a custom script

chaosmirage committed:
fix(nx-infra-plugin): correct nested submodule paths

chaosmirage committed:
fix(nx-infra-plugin): package scripts

chaosmirage committed:
fix(devextreme-react): package scripts

chaosmirage committed:
chore(nx-infra): build plugin on postinstall

chaosmirage committed:
chore(nx-infra-plugin): add custom build script for `nx-infra-plugin` to use postinstall hook

chaosmirage committed:
fix(nx-infra-plugin): resolve glob pattern failures on Windows

The glob library requires forward slashes as path separators across
all platforms, but path.join() creates backslashes on Windows. Glob
interprets backslashes as escape characters rather than path
separators, causing pattern matching to fail and executors to
process zero files on Windows.

Created normalizeGlobPathForWindows() utility function and applied
it to build-typescript and add-license-headers executors. Also
changed empty file list from warning to error for fail-fast
behavior.
